pandas/core/groupby/groupby.py

```diff
@@ -2135,7 +2134,6 @@ def sem(self, ddof: int = 1):
             result.iloc[:, result_ilocs] /= np.sqrt(counts.iloc[:, count_ilocs])
         return result
 
-    @final
```

Comment:
you could probably change `DataFrame | Series` to `NDFrameT` as this class is generic but would need to handle the `as_index` kwarg with overloads

Reply:
Given what I wrote above, it's not clear that I have to handle `as_index` as you can only apply that to a `DataFrame`, meaning you get a `DataFrame` as a result.

Reply:
Also, there are other methods in `groupby.py` that return `Series | DataFrame`, so I'd rather just keep this change minimal without introducing more things.

Reply:
as this class is already generic there should be no reason to add methods just to return the super result with a cast. Maybe add a comment to the added methods that they have been added for typing and to restore the `@final` once sorted.

Reply:
> as this class is already generic there should be no reason to add methods just to return the super result with a cast. Maybe add a comment to the added methods that they have been added for typing and to restore the `@final` once sorted.

OK, added comments in next commit